File: front-server/src/apis/response/chat/chat-room.response.dto.ts
import ResponseDto from '../response.dto';
import ChatMessageResponseDto from './chat-message.response.dto';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export default interface ChatRoomResponseDto extends ResponseDto {
  roomId: string;
  seller: string;
  lastSendDate: string;
  lastSendMessage: string;
  imageUrl: string;
  chatMessageList?: ChatMessageResponseDto[];
}

File: chat-service/src/main/java/com/wesell/chatservice/dto/response/ChatRoomResponseDto.java
package com.wesell.chatservice.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ChatRoomResponseDto {
    private String roomId;
    private String seller;
    private String consumer;
    private boolean isAlive;
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String lastSendDate;
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String lastSendMessage;
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String imageUrl;
}
